### PR TITLE
Add blog pagination

### PR DESCRIPTION
**Pull request by colebow**
Adding pagination to the blog so it's easier / more intuitive to find old blogs. Screenshots:

Page 1 still has the three most recent posts as big cards:
image

Then at the bottom, a "Page 1 of 10" on the left, a grayed out "Previous" button, and a "Next" button to view more blogs.
image

If we go to page 2, the three larger cards and the "Recent" header to separate them from additional posts disappears, and we're just paging through more blogs:
image

At the bottom, previous button lights up, page updates:
image

And when we get to the end, it works gracefully, and the next button grays itself out:
image

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -33,3 +33,6 @@ linkedin_audience: 2842796
 twitter_user: trinodb
 twitter_hashtag: trinodb
 youtube_channel: trinodb
+
+paginate: 12
```
Comment (bitsondatadev) [inline _config.yml:37]: So it's clear what this does:
> To enable pagination for posts on your blog, add a line to the _config.yml file that specifies how many items should be displayed per page:

```paginate: 5```

https://jekyllrb.com/docs/pagination/#enable-pagination

At first I thought this was total pages and we would have to keep updating the static value.

File: blog/archive.md [hunk 2]
```diff
@@ -29,10 +29,7 @@ title: Blog archives
         {% endif %} -->
       </div>
     {% endfor %}
-    <!--<hr />
-    <p class="overline">Want to stay up to date with the latest news from Strati? <a
-      href="#">Subscribe</a> to the blog!
-    </p>-->
+    <!--<hr /> -->
```
Comment (bitsondatadev) [inline blog/archive.md:32]: What is this doing? Can this line be entirely deleted?

File: blog/index.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
+---
```
Comment (bitsondatadev) [inline blog/index.html:1]: To be consistent with the other pages just keep this in .md file. I know pretty much all of this is html but the goal in general is to write as much in markdown as possible unless you're working with a template. These index pages are exceptions but should still ideally be as much markdown as possible.

**Review by bitsondatadev — APPROVED** [review]
Few nits but the buttons look sooooo good. Love the navigation.

File: blog/archive.md (hunk 2 above)
Comment (colebow) [reply blog/archive.md:32]: Yeah, I think so.

File: blog/archive.md [hunk 4]
```diff
@@ -29,10 +29,6 @@ title: Blog archives
         {% endif %} -->
       </div>
     {% endfor %}
-    <!--<hr />
```
Comment (mosabua) [inline blog/archive.md:32]: Hahah .. flashback

**Review by mosabua — APPROVED** [review]
VERY nice!

**Pull request merged**